core/src/main/java/org/apache/calcite/rel/rules/AggregateValuesRule.java

```diff
@@ -75,10 +91,18 @@ public AggregateValuesRule(RelBuilderFactory relBuilderFactory) {
   @Override public void onMatch(RelOptRuleCall call) {
     final Aggregate aggregate = call.rel(0);
     final Values values = call.rel(1);
-    Util.discard(values);
     final RelBuilder relBuilder = call.builder();
     final RexBuilder rexBuilder = relBuilder.getRexBuilder();
 
+    if ((aggregate.getGroupCount() != 0 || !values.getTuples().isEmpty())
```

Comment:
This judgment condition may have some problems. Maybe you can change like this.
```
    if (aggregate.getGroupCount() == 0 && values.getTuples().isEmpty()) {
      // old
    } else if (/* your condition */) {
      
    }
```
This may be a bad case:
```
    final String sql = "select count(*), sum(empno) from emp where false group by empno";
```